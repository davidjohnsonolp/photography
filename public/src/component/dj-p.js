import buildShadowRoot from '../lib/build-shadow-root.js';

class DJP extends HTMLElement {
  constructor() {
    super();
    const html = `
      <style>
        :host {
        }
        p {
          display: block;
          margin: 1em 0;
        }
      </style>
      <p>
        <slot></slot>
      </p>
    `;
    buildShadowRoot(html,this);
  }
}

customElements.define('dj-p', DJP);
export default DJP;
