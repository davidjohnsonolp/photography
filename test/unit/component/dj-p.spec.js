import '../../../public/src/component/dj-p.js';

let customP;

describe('DJP Custom Paragraph', () => {
  beforeEach(() => {
    customP = window.document.createElement('dj-p'); 
    document.body.appendChild(customP);
  });

  afterEach(() => {
    document.body.getElementsByTagName('dj-p')[0].remove();
  });

  it('renders correctly', () => {
    expect({ html: customP.shadowRoot.innerHTML }).toMatchSnapshot();
  });
});