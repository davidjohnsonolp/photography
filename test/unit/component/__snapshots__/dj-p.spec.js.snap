// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DJP Custom Paragraph renders correctly 1`] = `
Object {
  "html": "
      <style>
        :host {
        }
        p {
          display: block;
          margin: 1em 0;
        }
      </style>
      <p>
        <slot></slot>
      </p>
    ",
}
`;
